### PR TITLE
[feat]: [PL-26203]: defining DTOs for custom shell script template

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35050)

### CODE REVIEW

**Comment by shreyas-harness** [comment]
trigger checkstyle

**Comment by shreyas-harness** [comment]
trigger messagecheck

File: 870-orchestration/src/main/java/io/harness/engine/expressions/ExecutionTarget.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.data.annotation.TypeAlias;
+
+@Data
+@Builder
+@OwnedBy(PL)
+@TypeAlias("ScriptExecutionTarget")
+public class ExecutionTarget {
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> host;
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> connectorRef;
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) ParameterField<String> workingDirectory;
```
Comment (meenakshi-harness) [inline 870-orchestration/src/main/java/io/harness/engine/expressions/ExecutionTarget.java:28]: What is ParameterField? Why can't we keep these fields only as string?

File: 870-orchestration/src/main/java/io/harness/engine/expressions/CustomShellScriptSpec.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.TypeAlias;
+
+@Data
+@NoArgsConstructor
+@JsonTypeName(TemplateEntityConstants.SCRIPT)
+@TypeAlias("ShellScriptSpec")
+@OwnedBy(HarnessTeam.PL)
+public class CustomShellScriptSpec implements Visitable, WithDelegateSelector {
```
Comment (meenakshi-harness) [inline 870-orchestration/src/main/java/io/harness/engine/expressions/CustomShellScriptSpec.java:44]: Change its name to ScriptSpec

File: 870-orchestration/src/main/java/io/harness/engine/expressions/CustomShellScriptBaseSource.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Builder;
+import lombok.Data;
+import org.springframework.data.annotation.TypeAlias;
+
+@Data
+@Builder
+@JsonTypeName("Inline")
+@OwnedBy(PL)
+@TypeAlias("ShellScriptBaseSource")
+public class CustomShellScriptBaseSource {
```
Comment (meenakshi-harness) [inline 870-orchestration/src/main/java/io/harness/engine/expressions/CustomShellScriptBaseSource.java:27]: change name lets call all DTOs starting from Script...

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 9 Issues 
* Code issue 9 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35050)

**Pull request closed without merging**